README.md

```diff
@@ -214,6 +214,50 @@ fastify.register(require('fastify-swagger'), {
 ### swagger options
 Calling `fastify.swagger` will return to you a JSON object representing your api, if you pass `{ yaml: true }` to `fastify.swagger`, it will return you a yaml string.
 
+### Open API (OA) Parameter Options
+
+Note: OA's terminology differs from Fastify's. OA uses the term "parameter" to refer to those parts of a request that in [Fastify's validation documentation](https://www.fastify.io/docs/latest/Validation-and-Serialization/#validation) are called "querystring", "params", "headers".
+
+OA provides some options beyond those provided by the JSON schema specification for specifying the shape of parameters. A prime example of this the option for specifying how to encode those parameters that should be handled as arrays of values. There is no single universally accepted method for encoding such parameters appearing as part of query strings. OA2 provides a `collectionFormat` option which allows specifying how an array parameter should be encoded. (We're giving an example in the OA2 specification, as this is the default specification version used by this plugin. The same principles apply to OA3.) Specifying this option is easy. You just need to add it to the other options for the field you are defining. Like in this example:
+
+```
+fastify.route({
+  method: 'GET',
+  url: '/',
+  schema: {
+    querystring: {
+      type: 'object',
+      required: ['fields'],
+      additionalProperties: false,
+      properties: {
+        fields: {
+          type: 'array',
+          items: {
+            type: 'string'
+          },
+          minItems: 1,
+          //
+          // Note that this is an Open API version 2 configuration option.  The
+          // options changed in version 3.
```

Comment:
Good work! Which is the option for v3?

Reply:
I decided to mention it without going into the gory details so that *if* someone tries it in v3 the note gives them fair warning. I decided against going into more details mainly because:

1. I think the plugin currently does not let us use v3. Mind you, the way I came to this conclusion was that I shoved the v3 options meant to do the same thing as I was doing in v2 in my configuration and it did nothing good. Then I eyeballed the code and [saw](https://github.com/fastify/fastify-swagger/blob/70461eeaa5ea7290b1549e98b7c03f3ac1511a15/dynamic.js#L62):

   ```
    swaggerObject.swagger = '2.0'
   ```

   I scanned the code and did not see how this specific field might be overwritten somehow. Hmm... actually now that I review this, I'm thinking that having Swagger run in v3 mode would require more than changing the value of this field. As I recall, `swagger` is not a valid field in v3 at all. When running in v3 mode, it should set another field, the `openapi` field I believe.

   Also, I did not find mention in the plugin documentation as to how to select an Open API version. Switching versions is a big deal because v3 is not backward compatible with v2. So the lack of documentation was (for me) another indication that probably switching versions is not possible right now.
   
2. Going into the details of how it works in v3 would require more explanation than I think is warranted for the issue at hand here. For one thing, in v3 the configuration which is equivalent to v2's `collectionFormat: "multi"` *is the default handling in v3*. (The default in v2 was `collectionFormat: "csv"`.) So in fact, a lack of further specification in v3 gives a behavior which is compatible with the default query parser. Stated explicitly in v3's options, v3's default is becomes: `style: "form", explode: true`. Ok, how do these two interact?

   Basically, I'm trying to avoid turning the README into a tutorial on how to use Open API.

Reply:
Great writeup, can you mention that this module supports v2 and not v3 in the comment?

Reply:
Done.